File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l=0;
        int r=height.size()-1;
        int max=0;
        int area=0;
        while(l<r){
            area=min(height[l],height[r])*(r-l);
            if(height[l]<height[r]){
                l++;
            }
            else{
                r--;
            }
            if (max<area){
                max=area;
            }
        }
        return max;
    }
};

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        i=0
        j=len(height)-1
        most=-inf
        while i<j:
            curr=(j-i)*(min(height[i],height[j]))
            most=max(curr,most)
            
            if height[i]<height[j]:
                i+=1
            else:
                j-=1

        return most


